storage/innobase/fut/fut0lst.cc

```diff
@@ -50,7 +50,7 @@ flst_add_to_empty(
 	ut_ad(mtr_memo_contains_page_flagged(mtr, node,
 					     MTR_MEMO_PAGE_X_FIX
 					     | MTR_MEMO_PAGE_SX_FIX));
-	len = flst_get_len(base);
+	ut_d(len = flst_get_len(base));
 	ut_a(len == 0);
```

Comment:
Why not `ut_ad(!flst_get_len(base))`? OTOH I'm not complete sure if we can replace `ut_a` with `ut_ad`. In InnoDB it is common to handle errors with asserts in release builds. E.g. if this data is read from disk, I'd keep `ut_a`. But I'll leave it up to @dr-m to decide.

But anyway, please remove `len`, it looks useless.

Reply:
> In InnoDB it is common to handle errors with asserts in release builds.

I've thought that it wasn't desired behaviour to die in assert in release built.

Reply:
It is not desired behaviour definitely. But crashing later or corrupting data files is even worse.

There's distinction between asserts for debug purpose (`ut_ad`) and asserts for error handling purpose (`ut_a`). Unfortunately since recently MySQL/InnoDB started using `ut_a` for both purposes. I can't really tell which type of assert is this one.